### ISSUE TITLE
TextInput does not scroll with the offscreen text on iOS with certain padding and heights

### ISSUE THREAD

**Issue by jkdf2** [facebook/react-native#14711]
### Description

The TextInput component, given certain padding and heights, does not scroll as the user types into the rightmost edge of the screen on iOS.

![screen shot 2017-06-23 at 4 26 09 pm](https://user-images.githubusercontent.com/2641739/27501044-b1631dfc-5830-11e7-826c-45b58ce6faed.png)

### Reproduction Steps

See the Snack for an example height, padding, and fontSize where this issue occurs. 

#### Sample Code

[Relevant snack](https://snack.expo.io/rk_9KgsmW).

### Additional Information

* React Native version: react-native@0.45.1
* Platform: iOS
* Build tools: Snack SDK is v16.0.0

**Comment by jkdf2** [comment]
Confirmed still an issue in `react-native@0.46.0`.

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by ArcadeRenegade** [comment]
Any workarounds?

**Comment by eliperkins** [comment]
Reproduced on react-native 0.49.1.

I've also reproduced without using any padding, but rather just constraining `height`.

After applying an *even* `padding` value, the overflow and cursor follow each other.

I have a feeling this is related to fractional values, given that things seem to work with event numbers, but I'm not sure where to look. Any guidance on where, within React Native, to debug this would be greatly appreciated and I'll work on a PR to follow!
